setup.py

```diff
@@ -11,76 +11,68 @@
 
 import os
-from distutils.core import *
-from subprocess import *
+import platform
+import subprocess
 
-package_name = 'Tkrzw'
+from setuptools import Extension, setup
+
+package_name = 'tkrzw'
 package_version = '0.1'
 package_description = 'a set of implementations of DBM'
 package_author = 'Mikio Hirabayashi'
 package_author_email = 'hirarin@gmail.com'
 package_url = 'http://dbmx.net/tkrzw/'
 module_name = 'tkrzw'
 
-ldpath = os.environ.get('LD_LIBRARY_PATH')
-ldpaths = []
-if ldpath:
-    ldpaths.append(ldpath)
-ldpaths.append('/usr/local/lib')
-os.environ['LD_LIBRARY_PATH'] = ':'.join(ldpaths)
+extra_compile_args = ["-std=c++17", "-Wall"]
+sources = ['tkrzw.cc']
+
+ld_paths = [
+    '/usr/local/lib'
+]
+os.environ['LD_LIBRARY_PATH'] = os.environ.get('LD_LIBRARY_PATH', '') + ':'.join(ld_paths)
 
-def getcmdout(cmdargs):
-    try:
-        pipe = Popen(cmdargs, stdout=PIPE)
-        output = pipe.communicate()[0].decode('utf-8')
-    except:
-        output = ""
-    return output.strip()
 
-include_dirs = []
-myincopts = getcmdout(['tkrzw_build_util', 'config', '-i']).split()
-for incopt in myincopts:
-    if incopt.startswith('-I'):
-        incdir = incopt[2:]
-        include_dirs.append(incdir)
-if len(include_dirs) < 1:
-    include_dirs = ['/usr/local/include']
+def parse_build_flags(cmd_args):
+    result = {}
+    output = subprocess.check_output(cmd_args).strip().decode()
+    for item in output.split():
+        flag = item[0:2]
+        result.setdefault(flag, []).append(item[2:])
+    return result
 
-extra_compile_args = ["-std=c++17", "-Wall"]
-sources = ['tkrzw.cc']
 
-library_dirs = []
-libraries = []
-mylibopts = getcmdout(['tkrzw_build_util', 'config', '-l']).split()
-for libopt in mylibopts:
-    if libopt.startswith('-L'):
-        libdir = libopt[2:]
-        library_dirs.append(libdir)
-    elif libopt.startswith('-l'):
-        libname = libopt[2:]
-        libraries.append(libname)
-if len(library_dirs) < 1:
-    library_dirs = ['/usr/local/lib']
-if len(libraries) < 1:
-    if (os.uname()[0] == "Darwin"):
-        libraries = ['tkrzw', 'stdc++', 'pthread', 'm', 'c']
-    else:
-        libraries = ['tkrzw', 'stdc++', 'rt', 'pthread', 'm', 'c']
+# Parse include dirs
+flags = parse_build_flags(['tkrzw_build_util', 'config', '-i'])
+include_dirs = flags.get('-I', []) or ['/usr/local/include']
 
-module = Extension(module_name,
-                   include_dirs = include_dirs,
-                   extra_compile_args = extra_compile_args,
-                   sources = sources,
-                   library_dirs = library_dirs,
-                   libraries = libraries)
+# Parse library dirs and libraries
+flags = parse_build_flags(['tkrzw_build_util', 'config', '-l'])
+library_dirs = flags.get('-L', []) or ['/usr/local/lib']
+libraries = flags.get('-l', [])
+if not libraries:
+    libraries = ['tkrzw', 'stdc++', 'pthread', 'm', 'c']
+    if platform.system() != "Darwin":
+        libraries.merge(['rt', 'atomic'])
```

Comment:
In principle, underlying libraries should be listed from higher-level to lower level, according to their dependency.
Of them, tkrzw is the highest and the stdc++ is the second highest.  rt, atomic, and pthread comes next.  Then, m, and c as the lowest.
Therefore I prefer to list them as-is wholly for each of non-Darwin and Darwin systems.
BTW, this logic exists just in case that "tkrzw_build_util" is separated from the binary package of the core library.

Reply:
Ok, I’ll remove that part. By the way, simply switching the import to setuptools works out of the box. So I can remove the refactoring if you prefer.
(Internet down right now, will update tomorrow)